server.go

```diff
@@ -127,8 +128,9 @@ func (s *Server) Start() error {
 	// TODO(roasbeef): more granular macaroons, can ask user to make just
 	// what we need
 	s.lndClient, err = lndclient.NewBasicClient(
-		s.cfg.Lnd.Host, s.cfg.Lnd.TLSPath, s.cfg.Lnd.MacaroonDir,
-		s.cfg.Network,
+		s.cfg.Lnd.Host, s.cfg.Lnd.TLSPath,
+		path.Dir(s.cfg.Lnd.MacaroonPath), s.cfg.Network,
```

Comment:
probably a `lndclient` comment really, but shouldn't we really be passing in the full macaroon path here also, and remove the dir?

Reply:
True. We should try to get rid of the base client anyway. I put this on my TODO list.

Reply:
The `lndclient.MacFilename` functinoal arg was initially meant for this purpose...but seems that knowledge wasn't properly passed down to those that modify the library most frequently.